File: src/VBNetFx/Controls/ADNullableValueSubControls/ADEditableUpDownValueForNullableValueControl.vb
Imports System.Windows.Forms

Friend Class ADEditableUpDownValueForNullableValueControl
    Inherits ADEditableValueForNullableValueControlTemplate(Of Integer)

    Protected WithEvents myNumericUpDown As NumericUpDown

    Sub New()
        MyBase.New()
        myNumericUpDown = New NumericUpDown
        myControl = myNumericUpDown
        myControl.Name = "ADVariantTextControl" & myInstanceCounter
    End Sub

    Public Overrides Property Size() As System.Drawing.Size
        Get
            Return myControl.Size
        End Get
        Set(ByVal Value As System.Drawing.Size)
            myControl.Size = Value
        End Set
    End Property

    Public Overrides Function MeasureHeight() As Integer
        Return (myControl.Height) * -1
    End Function

    Public Overrides Property Value() As Object
        Get
            Return DirectCast(myControl, NumericUpDown).Text
        End Get
        Set(ByVal Value As Object)
            If Value Is Nothing Then
                DirectCast(myControl, NumericUpDown).Text = ""
            Else
                DirectCast(myControl, NumericUpDown).Text = Value.ToString()
            End If
        End Set
    End Property

    Private Sub myNumericUpDown_TextChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles myNumericUpDown.TextChanged
        OnValueChanged(e)
    End Sub

    Private Sub myNumericUpDown_Validated(ByVal sender As Object, ByVal e As System.EventArgs) Handles myNumericUpDown.Validated
        OnValidated(e)
    End Sub

    Private Sub myNumericUpDown_Validating(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles myNumericUpDown.Validating
        OnValidating(e)
    End Sub
End Class

File: src/NetFxCSharp/Controls/ADNullableValueSubControls/ADEditableUpDownValueForNullableValueControl.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace ActiveDev.Controls
{
    internal class ADEditableUpDownValueForNullableValueControl : ADEditableValueForNullableValueControlTemplate<int>
    {
        private NumericUpDown _myNumericUpDown;
        protected NumericUpDown myNumericUpDown
        {
            get
            {
                return _myNumericUpDown;
            }

            set
            {
                if (_myNumericUpDown != null)
                {
                    _myNumericUpDown.TextChanged -= myNumericUpDown_TextChanged;
                    _myNumericUpDown.Validated -= myNumericUpDown_Validated;
                    _myNumericUpDown.Validating -= myNumericUpDown_Validating;
                }

                _myNumericUpDown = value;
                if (_myNumericUpDown != null)
                {
                    _myNumericUpDown.TextChanged += myNumericUpDown_TextChanged;
                    _myNumericUpDown.Validated += myNumericUpDown_Validated;
                    _myNumericUpDown.Validating += myNumericUpDown_Validating;
                }
            }
        }

        public ADEditableUpDownValueForNullableValueControl() : base()
        {
            myNumericUpDown = new NumericUpDown();
            myControl = myNumericUpDown;
            myControl.Name = "ADVariantTextControl" + myInstanceCounter;
        }

        public override System.Drawing.Size Size
        {
            get
            {
                return myControl.Size;
            }

            set
            {
                System.Drawing.Size Value = value;
                myControl.Size = Value;
            }
        }

        public override int MeasureHeight()
        {
            return (myControl.Height) * -1;
        }

        public override object Value
        {
            get
            {
                return ((NumericUpDown)myControl).Text;
            }

            set
            {
                object Value = value;
                if (Value == null)
                {
                    ((NumericUpDown)myControl).Text = "";
                }
                else
                {
                    ((NumericUpDown)myControl).Text = Value.ToString();
                }
            }
        }

        private void myNumericUpDown_TextChanged(object sender, System.EventArgs e)
        {
            OnValueChanged(e);
        }

        private void myNumericUpDown_Validated(object sender, System.EventArgs e)
        {
            OnValidated(e);
        }

        private void myNumericUpDown_Validating(object sender, System.ComponentModel.CancelEventArgs e)
        {
            OnValidating(e);
        }
    }
}